### ISSUE TITLE
Support for arbiter/temporary master node

### ISSUE THREAD

**Issue by vbohata** [elastic/elasticsearch#32462]
It would be great to be able to make one of master nodes arbiter/temporary master. We have 3 independent locations - 2 fast location and 1 slow location. It is not a good idea to have a active master node in the slow location but in case of outage of the fast location it would be great to have one in this slow location.

So I would like to be able to configure something like: make me master only if necessary and in case of 3 master-eligible nodes available give it up and let one of the master nodes in fast location drive the cluster.

Btw. in this case it will be never required to be master-node in slow location active. So something like "quorum_only" or "arbiter_only" config option would be great.

**Comment by DaveCTurner** [comment]
Duplicate of #14340. It's tricky, and probably depends on #32006.
